src/workers/query_worker.cpp

```diff
@@ -182,9 +183,14 @@ void query_worker::query(zmq::socket& dealer)
 
 // Class and method names must match protocol expectations (do not change).
 #define ATTACH(class_name, method_name, node) \
+attach(#class_name "." #method_name, \
+    std::bind(&bc::server::class_name::method_name, \
+        std::ref(node), _1, _2));
+
+#define ATTACH_SETTINGS(class_name, method_name, node) \
     attach(#class_name "." #method_name, \
         std::bind(&bc::server::class_name::method_name, \
-            std::ref(node), _1, _2));
+            std::ref(node), _1, _2, bitcoin_settings_));
```

Comment:
Settings should precede placeholders here, though this macro should not be required (see below).

Reply:
Because the node is passed there should not probably be a need to pass settings. The node should probably expose bitcoin_settings() as it dies others. We may want to create a settings (config) object to expose all settings in node, but for now a distinct method is sufficient.

Reply:
Yeah, I keep neglecting that `bitcoin_settings()` is implemented in `node`. Done.